st2actions/st2actions/notifier/notifier.py

```diff
@@ -79,29 +76,27 @@ def __init__(self, connection, queues, trigger_dispatcher=None):
     def process(self, execution_db):
         execution_id = str(execution_db.id)
         extra = {'execution': execution_db}
-        LOG.debug('Processing execution %s', execution_id, extra=extra)
-
-        if ('orquesta' in execution_db.context and
-                execution_db.status == LIVEACTION_STATUS_PAUSED):
-            wf_svc.handle_action_execution_pause(execution_db)
+        LOG.debug('Processing action execution "%s".', execution_id, extra=extra)
 
         if execution_db.status not in LIVEACTION_COMPLETED_STATES:
-            LOG.debug('Skipping processing of execution %s since it\'s not in a completed state' %
-                      (execution_id), extra=extra)
+            msg = 'Skip action execution "%s" because state "%s" is not in a completed state.'
+            LOG.debug(msg % (str(execution_db.id), execution_db.status), extra=extra)
             return
 
-        liveaction_id = execution_db.liveaction['id']
-        liveaction_db = LiveAction.get_by_id(liveaction_id)
-        self._apply_post_run_policies(liveaction_db=liveaction_db)
+        # Get the corresponding liveaction record.
+        liveaction_db = LiveAction.get_by_id(execution_db.liveaction['id'])
+
+        # If the action execution is executed under an orquesta workflow, policies for the
+        # action execution will be applied by the workflow engine. A policy may affect the
+        # final state of the action execution thereby impacting the state of the workflow.
+        if 'orquesta' not in execution_db.context:
```

Comment:
I assume there is no better way to detect if it's an orquesta worklow.

Perhaps it would still be better to refactor it in utility function since it's used in multiple places - `is_execution_orquesta_workflow` or similar.

Reply:
There's no better way to detect if the action execution is executed as a task of an orquesta workflow.  I can add a helper method.